main/src/library/Graph.flix

```diff
@@ -78,4 +79,16 @@ namespace Graph {
     pub def isConnected(src: {src :: t}, dst: {dst :: t}, g: m[(t, t)]): Bool with Foldable[m], Boxable[t] =
         reachable(src.src, g) |> Set.exists(x -> dst.dst == x)
 
+    ///
+    /// Returns the strongly connected components of the directed graph `g`.
+    ///
+    pub def stronglyConnectedComponents(g: m[(t, t)]): List[Set[t]] with Foldable[m], Boxable[t] =
+        let edges = inject g into Edge;
+        let connected = #{
+            // If `n1` can reach `n2` and `n2` can reach `n1`
+            // then they are part of the same strongly connected component.
+            Connected(n1; Set#{n2}) :- Reachable(n1, n2), Reachable(n2, n1).
+        };
+        query edges, nodes(), reachability(), connected select x from Connected(_; x)
```

Comment:
When I query it like this and do not care about the first argument of `Connected(_; x)`, then I only get one instance of each strongly connected component. Is this slow?

Reply:
Oh yeah this is nice, I didn't think of that. I think that this is the same as the set of sets solution though I'm not sure. I will test the speed in a bit.

Reply:
Using this rule to query from `Components(s) :- fix Connected(n; s), if Some(n) == Set.minimum(s).` seems a good bit faster (both using a fully disconnected graph and a fully connected graph)